python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -179,6 +211,57 @@ def getURI(
         # Docstring inherited.
         raise NotImplementedError()
 
+    def get_dataset_type(self, name: str) -> DatasetType:
+        # In future implementation this should directly access the cache
+        # and only go to the server if the dataset type is not known.
+        path = f"dataset_type/{name}"
+        response = self._client.get(self._get_url(path))
+        response.raise_for_status()
+        return DatasetType.from_simple(SerializedDatasetType(**response.json()), universe=self.dimensions)
+
+    def get_dataset(self, id: DatasetId) -> DatasetRef | None:
+        path = f"dataset/{id}"
+        response = self._client.get(self._get_url(path))
+        if response.json() is None:
+            return None
+        return DatasetRef.from_simple(SerializedDatasetRef(**response.json()), universe=self.dimensions)
+
+    def find_dataset(
+        self,
+        datasetType: DatasetType | str,
+        dataId: DataId | None = None,
+        *,
+        collections: CollectionArgType | None = None,
+        timespan: Timespan | None = None,
+        datastore_records: bool = False,
+        **kwargs: Any,
+    ) -> DatasetRef | None:
+        if collections is None:
+            if not self.collections:
+                raise NoDefaultCollectionError(
+                    "No collections provided to find_dataset, and no defaults from butler construction."
+                )
+            collections = self.collections
+        # Temporary hack. Assume strings for collections. In future
+        # want to construct CollectionWildcard and filter it through collection
+        # cache to generate list of collection names.
+        wildcards = CollectionWildcard.from_expression(collections)
```

Comment:
Maybe do a quick check here and raise if `patterns` specifies something?

Reply:
Seems like I have to change the API to something a bit simpler for collection list anyhow.